### PR TITLE
Refactored utf16->utf8 support from DynDep to common

### PR DESCRIPTION
**Pull request by DrusTheAxe**
Moved DynamicDependency's `to_utf8` functions to `Microsoft.Utf8.h` (and renamed `ToUtf8`) as the complementary inverse of the `ToUtf16` functions already there.

Deleted DynDep's now-obsolete `utf8.h` and changed to the `Microsoft.Utf8` namespace functions. All tests pass as expected

### CODE REVIEW

**Review by hulumane — APPROVED** [review]

File: dev/Common/Microsoft.Utf8.h [hunk 1]
```diff
@@ -45,8 +49,52 @@ template<> inline winrt::hstring ToUtf16(PCSTR utf8)
...
+{
+    return ToUtf16(utf8.c_str());
+}
+
 inline winrt::hstring ToHString(PCSTR utf8)
 {
     return ToUtf16<winrt::hstring>(utf8);
 }
+
+// UTF16->UTF8 conversions
```
Comment (asklar) [inline dev/Common/Microsoft.Utf8.h:62]: @DrusTheAxe  could you clarify why are all these needed / why don't we use `winrt::to_hstring` and `winrt::to_string` ?

**Pull request merged**